### ISSUE TITLE
Use Spack in Docker container?

### ISSUE THREAD

**Issue by nilsvu** [sxs-collaboration/spectre#2485]
Should we use Spack to install dependencies in the Docker container? This could simplify the Dockerfile quite a lot and it would give an example how to use Spack to install the dependencies. Any disadvantages?

**Comment by fmahebert** [comment]
The Dockerfile has the answer (or at least, part of the answer) to this question...
>  We install dependencies not available through apt manually rather than using Spack since Spack ends up building a lot of dependencies from scratch that we don't need. Thus, not building the deps with Spack reduces total build time of the Docker image.

**Comment by nilsvu** [comment]
Right, Spack installing everything from source when the system has perfectly fine packages installed can be annoying. More recent versions of Spack improve on that quite a lot though, e.g. by [automatically finding external packages](https://spack.readthedocs.io/en/latest/build_settings.html#automatically-find-external-packages) (basically run `spack external find --not-buildable` to auto-detect and not build system-installed packages) and by using [environments](https://spack.readthedocs.io/en/latest/environments.html). There's also a new version of the Spack concretizer in the works that will supposedly be less eager to re-build everything, but that's to be seen.

**Comment by nilsdeppe** [comment]
No, not for several years. I don't trust Spack to be sufficiently stable that this won't cause an enormous amount of work every time I try to update the container. Major headaches I've had are:
- specific Spack versions might not have the newest package version that we need
- using Spack develop means random things break all the time
- Spack does not release frequently enough and releases are only somewhat stable
- it's actually easier to build libraries from scratch than to deal with convincing Spack not to install everything under the sun. While this might be improved, I don't really feel like beta testing this because it'll waste a lot of my time.Basically, major container size bloat is difficult to manage
- there are almost certainly things not available in Spack, then we have to add them and maintain them in Spack. Again, more work than just bumping a version number in our container.

In short, since I've been the one maintaining the container: no because it'll very likely waste a lot of time and offers basically no benefit over what we do right now.

Is that enough to close this issue?
